File: back/model/brand.go
package model

import (
	"time"

)

// Brand 品牌
type Brand struct {
	ID          string    `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`            // 品牌ID
	Slug        string    `gorm:"type:varchar(50);not null;uniqueIndex" json:"slug"`                    // 品牌Slug
	Name        string    `gorm:"type:varchar(50);not null;uniqueIndex" json:"name"`                    // 品牌名称
	Description string    `gorm:"type:text" json:"description"`                                         // 品牌描述
	Website     *string   `gorm:"type:varchar(255)" json:"website,omitempty"`                           // 品牌网站
	Logo        string    `gorm:"type:varchar(255)" json:"logo"`                                        // 品牌LOGO
	SortOrder   int       `gorm:"default:0;index" json:"sortOrder"`                                     // 排序顺序
	CreatedAt   time.Time `gorm:"not null" json:"createdAt"`                                           // 创建时间
	UpdatedAt   time.Time `gorm:"not null" json:"updatedAt"`                                           // 更新时间

	Products []Product `gorm:"foreignKey:BrandID;constraint:OnDelete:RESTRICT" json:"products,omitempty"` // 关联的产品
}

File: back/web/src/types/brand.ts
// 品牌实体
export interface Brand {
  id: string;
  name: string;
  description?: string;
  logo?: string;
  website?: string;
  sortOrder: number;
  createdAt: string;
  updatedAt: string;
}

// 创建品牌请求
export interface CreateBrandRequest {
  name: string;
  description?: string;
  logo?: string;
  website?: string;
  sortOrder: number;
}

// 更新品牌请求
export interface UpdateBrandRequest {
  name?: string;
  description?: string;
  logo?: string;
  website?: string;
  sortOrder?: number;
}

// 品牌查询参数
export interface BrandQueryParams {
  page?: number;
  pageSize?: number;
  keyword?: string;
  sortBy?: 'updatedAt' | '';
  sortOrder?: 'asc' | 'desc';
}
